File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  let newArry= [];
  for(let i=0; i<= str.length; i++){
    newArry.unshift(str[i]);
  }
  let newStr= newArry.join("");
  return newStr
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// create empty array var to keep the new array
// iterate the string length
// unshift the i onto new array 
// shift removes dont forget
// revert array back to string

// And a written explanation of your solution

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  reversed_str= ""
  str.chars.each do |char|
    reversed_str= char + reversed_str
  end
  reversed_str

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# create an empty variable to store string in
# iterate through the string
# add the iteration of the element onto beg. of str element and est = to OG empty var


# And a written explanation of your solution